java/src/main/java/com/azure/graph/bulk/impl/impl/CosmosDBSQLBulkExecutor.java

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.graph.bulk.impl.impl;
+
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.azure.cosmos.models.CosmosBulkOperationResponse;
+import com.azure.cosmos.models.CosmosBulkOperations;
+import com.azure.cosmos.models.CosmosItemOperation;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.graph.bulk.impl.GraphBulkExecutor;
+import com.azure.graph.bulk.impl.ObjectToEdge;
+import com.azure.graph.bulk.impl.ObjectToVertex;
+import com.azure.graph.bulk.impl.model.GremlinEdge;
+import com.azure.graph.bulk.impl.model.GremlinVertex;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Flux;
+
+import java.util.ArrayList;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+@Slf4j
+public class CosmosDBSQLBulkExecutor<V, E> implements GraphBulkExecutor<V, E> {
```

Comment:
What is the lifetime of this instance? Can it be re-used for multiple bulk execute calls?

Reply:
The client this was used for re-used it (was a singleton bean), but applied the same policies for the entire solution. But that's a specific implementation, I like moving it to a parameter as you suggest though. Greater flexibility.

Reply:
As a design note: Long term I can see the bulk executor supporting a pattern where each Gremlin item op can be defined much like the Cosmos item ops are, eg.
```
graphBulkExecutor.excecuteOperations(Flux.just(
    GraphBulkOperations.getCreateVertexOperation(vertexA),
    GraphBulkOperations.getUpsertVertexOperation(vertexB),
    GraphBulkOperations.getCreateEdgeOperation(edgeAToB),
    GraphBulkOperations.getReplaceVertexOperation(vertexC),
    GraphBulkOperations.getDeleteVertexOperation(vertexD),
    GraphBulkOperations.getPatchVertexOperation(partialVertexE),
  ...)).blockLast();
```

Given that the initial version will only support single operation type per invocation, it might make sense to name the methods matching those op types and reserve the more general method name for when we can support heterogeneous operations in a single invocation.

For example, the methods we support today can be `executeCreateOperations(...)`, `executeUpsertOperations(...)`

Reply:
Pretty impactful refactor implementing this one. I moved the call to the containers execute into the sample code and focused the functionality on creating the operations.

This solves that leaky abstraction quite well, as I didn't want to hide the results of the containers execution of the bulk operation from the client implementation. Abstracting that call was never really a huge value add of this sample. 

Since it's no longer responsible for making the execution, I renamed the class to GremlinDocumentOperationCreator. Not 100% happy with that, quite wordy. 

It does push the responsibility of combining the operations back to the client, but I think that flows well with your recommendation above.

Reply:
I'll need to make a pass over the documentation to make sure the refactor is reflected correctly.

Reply:
I didn't necessarily mean for this to be done in this change, hence the 'long term'. Main concern was to keep the API open to having different invocation paths for uniform/non-uniform operation patterns.

But if you made the changes now that's great!